scipy/stats/tests/test_stats.py

```diff
@@ -6149,7 +6138,7 @@ def test_ttest_ind_axis_size_zero(self, b, expected_shape, xp):
         # The results should be arrays containing nan with shape
         # given by the broadcast nonaxis dimensions.
         a = xp.empty((3, 1, 0))
-        b = xp.asarray(b)
+        b = xp.asarray(b, dtype=a.dtype)
```

Comment:
When SCIPY_DEFAULT_DTYPE=float32, this was float64. The return value was, incorrectly float32 instead of float64 so the test passed anyway.